core/xchain/entitlement/entitlement.go

```diff
@@ -88,26 +130,39 @@ func (e *Evaluator) evaluateOrOperation(
 	defer rightCancel()
 	go func() {
 		leftResult, leftErr = e.evaluateOp(leftCtx, op.LeftOperation, linkedWallets)
-		if leftResult || leftErr != nil {
-			// cancel the other goroutine
-			// if the left result is true or there is an error
+		if leftResult {
+			// cancel the other goroutine if the left result is true, since we know
+			// the user is unentitled
 			rightCancel()
 		}
 		wg.Done()
 	}()
 
 	go func() {
 		rightResult, rightErr = e.evaluateOp(rightCtx, op.RightOperation, linkedWallets)
-		if rightResult || rightErr != nil {
-			// cancel the other goroutine
-			// if the right result is true or there is an error
+		if rightResult {
+			// cancel the other goroutine if the right result is true, since we know
+			// the user is entitled
 			leftCancel()
 		}
 		wg.Done()
 	}()
 
 	wg.Wait()
-	return leftResult || rightResult, nil
+	// If at least one child evaluates as entitled, log any errors and return a true result.
+	if leftResult || rightResult {
+		return true, nil
+	} else {
+		if leftErr != nil && rightErr != nil {
+			return false, fmt.Errorf("%w; %w", leftErr, rightErr)
+		} else {
+			finalErr := leftErr
+			if rightErr != nil {
+				finalErr = rightErr
+			}
+			return false, finalErr
```

Comment:
The and operation logic checks if an error was the result of the context cancelled. Should that be checked here as well?

Reply:
Yes, I think so. Probably I just didn't catch this because it didn't occur when I was running tests.